File: services/lucy-dotnet/src/Lucy.Auth.Api/Entities/AvatarPersona.cs
namespace Lucy.Auth.Api.Entities;

public sealed class AvatarPersona
{
    public string UserId { get; set; } = default!;
    public required string DisplayName { get; set; }
    public string? AvatarUrl { get; set; }
    public int IsAnonymous { get; set; } = 1;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public User User { get; set; } = null!;
}

File: lucy-web-stack/lucy-backend/src/main/java/com/lucy/backend/auth/entity/AvatarPersona.java
package com.lucy.backend.auth.entity;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Data
@Table(name = "AvatarPersonas")
public class AvatarPersona {
    
    @Id
    @Column(name = "UserId", length = 50)
    private String userId;

    @Column(name = "DisplayName", length = 50, nullable = false)
    private String displayName;

    @Column(name = "AvatarUrl", length = 255)
    private String avatarUrl;

    @Column(name = "IsAnonymous", nullable = false)
    private int isAnonymous = 1;
}
